renderer/containers/editor.js

```diff
@@ -20,7 +20,17 @@ export default class EditorContainer extends Container {
     const src = `file://${filePath}`;
     this.finishLoading = resolve;
 
-    this.setState({src, filePath, originalFilePath, fps, originalFps: fps, wasMuted: false, isNewRecording});
+    this.setState({
+      src,
+      filePath,
+      originalFilePath,
+      // TODO: Fix this ESLint violation
+      // eslint-disable-next-line react/no-access-state-in-setstate
```

Comment:
`setState` supports a callback which is meant for modifying the current state.

Reply:
unstated version 1 didn't support this. Version 2.x supports it but when I try to upgrade the cropper breaks. Moving the state changes to being async in 2.x, means that there's a ton of changes needed in the cropper logic, with a lot of `await`s for `setState` to make sure the timing works for resizing/moving. I have to convert the entire container to the new format, but it'll take a while and should probably be in a new PR

Reply:
Ah ok. No point in upgrading to unstated 2 though, as there is: https://github.com/jamiebuilds/unstated-next We should stay with unstated 1 for now and we can look at a hook-based replacement at some point when the state hook ecosystem is more mature.

Reply:
Sounds good. Might look into converting the React code to use hooks at some point. =

Also, we could look into converting at least the renderer code in TypeScript, since Next.js supports it out of the box with 0 config. Not sure if that makes more or less friendly for contributors

Reply:
Slowly converting to hooks would be super nice. I think that would help lower the threshold for contribution as hooks are more easily approachable.

Reply:
Yeah, TS would be nice, but I would wait at least a year. The current React type definition for TS is pretty subpar, and TS is annoying to deal with and way too many bugs.